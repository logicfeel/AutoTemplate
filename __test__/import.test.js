const fs        = require("fs");
const path      = require("path");
const AutoTask  = require("../src/auto-task").AutoTask;
const dirname   = path.join(__dirname, "/import/mod1");

/**
 * - 강제초기화
 * - 출판
 *  - 
 */

describe("task :: clear", () => {
    it("[ 생성 및 do_clear(1) ]", () => {
        autoTask = AutoTask.create(dirname);
        autoTask.isLog = false;
        autoTask.do_clear(1);   // 강제 클리어
    });

    it("- 파일 유무 : src/one.html (X)", () => {
        const fullPath = path.join(dirname, "src/one.html");
        expect(fs.existsSync(fullPath)).toBeFalsy();
    });
    
    describe("< hbs 내부에서 생성한 파일 >", ()=>{
        it("- 파일 유무 : src/inline/all/pre_p1.html (X)", () => {
            const fullPath = path.join(dirname, "src/inline/all/pre_p1.html");
            expect(fs.existsSync(fullPath)).toBeFalsy();
        });
        it("- 파일 유무 : src/inline/all/pre_p2.html (X)", () => {
            const fullPath = path.join(dirname, "src/inline/all/pre_p2.html");
            expect(fs.existsSync(fullPath)).toBeFalsy();
        });
        it("- 파일 유무 : src/inline/all/pre_p3.html (X)", () => {
            const fullPath = path.join(dirname, "src/inline/all/pre_p3.html");
            expect(fs.existsSync(fullPath)).toBeFalsy();
        });
        it("- 파일 유무 : src/inline/arg1_1/arg1_2/group_arg2/pre_p1_suf.html (X)", () => {
            const fullPath = path.join(dirname, "src/inline/arg1_1/arg1_2/group_arg2/pre_p1_suf.html");
            expect(fs.existsSync(fullPath)).toBeFalsy();
        });
        it("- 파일 유무 : src/inline/arg1_1/arg1_2/group_arg2/pre_p2_suf.html (X)", () => {
            const fullPath = path.join(dirname, "src/inline/arg1_1/arg1_2/group_arg2/pre_p2_suf.html");
            expect(fs.existsSync(fullPath)).toBeFalsy();
        });
        it("- 파일 유무 : src/inline/new_p1.html (X)", () => {
            const fullPath = path.join(dirname, "src/inline/new_p1.html");
            expect(fs.existsSync(fullPath)).toBeFalsy();
        });
        it("- 파일 유무 : src/inline2/def1/group_def2/p1.html (X)", () => {
            const fullPath = path.join(dirname, "src/inline2/def1/group_def2/p1.html");
            expect(fs.existsSync(fullPath)).toBeFalsy();
        });
        it("- 파일 유무 : src/inline2/def1/group_def2/p2.html (X)", () => {
            const fullPath = path.join(dirname, "src/inline2/def1/group_def2/p2.html");
            expect(fs.existsSync(fullPath)).toBeFalsy();
        });
    });
    
    describe("< ready() 에서 생성한 파일 >", ()=>{
        it("- 파일 유무 : src/ready/pre_p1_suf.html (X)", () => {
            const fullPath = path.join(dirname, "src/ready/pre_p1_suf.html");
            expect(fs.existsSync(fullPath)).toBeFalsy();
        });
        it("- 파일 유무 : src/ready/pre_p2_suf.html (X)", () => {
            const fullPath = path.join(dirname, "src/ready/pre_p2_suf.html");
            expect(fs.existsSync(fullPath)).toBeFalsy();
        });
        it("- 파일 유무 : src/ready/pre_p3_suf.html (X)", () => {
            const fullPath = path.join(dirname, "src/ready/pre_p3_suf.html");
            expect(fs.existsSync(fullPath)).toBeFalsy();
        });
        it("- 파일 유무 : src/ready/arg1/group_arg2/pre_p1_suf.html (X)", () => {
            const fullPath = path.join(dirname, "src/ready/arg1/group_arg2/pre_p1_suf.html");
            expect(fs.existsSync(fullPath)).toBeFalsy();
        });
        it("- 파일 유무 : src/ready/arg1/group_arg2/pre_p2_suf.html (X)", () => {
            const fullPath = path.join(dirname, "src/ready/arg1/pre_p2_suf/pre_p2_suf.html");
            expect(fs.existsSync(fullPath)).toBeFalsy();
        });
        it("- 파일 유무 : src/ready/new_p3.html (X)", () => {
            const fullPath = path.join(dirname, "src/ready/new_p3.html");
            expect(fs.existsSync(fullPath)).toBeFalsy();
        });
        it("- 파일 유무 : src/p3.html (X)", () => {
            const fullPath = path.join(dirname, "src/p3.html");
            expect(fs.existsSync(fullPath)).toBeFalsy();
        });
    });
});

describe("task :: publish", () => {
    beforeAll(() => {
        jest.resetModules();
    });

    it("[ 생성 및 do_publish() ]", () => {
        autoTask = AutoTask.create(dirname);
        autoTask.isLog = false;
        autoTask.do_publish();
    });
    
    describe("< *.hbs 검사 >", ()=>{
        it("- 파일 유무 : src/one.html", () => {
            const fullPath = path.join(dirname, "/src/one.html");
            expect(fs.existsSync(fullPath)).toBeTruthy();
        });
        it("- 파일 비교 : src/one.html", () => {
            const fullPath = path.join(dirname, "/src/one.html");
            const data = fs.readFileSync(fullPath,"utf-8");
            expect(fs.existsSync(fullPath)).toBeTruthy();
            expect(data).toMatchSnapshot();
        });
    });
    
    describe("< *.hbs inline 생성파일 검사 >", ()=>{
        it("- 파일 유무 : src/inline/all/pre_p1.html", () => {
            const fullPath = path.join(dirname, "src/inline/all/pre_p1.html");
            expect(fs.existsSync(fullPath)).toBeTruthy();
        });
        it("- 파일 유무 : src/inline/all/pre_p2.html", () => {
            const fullPath = path.join(dirname, "src/inline/all/pre_p2.html");
            expect(fs.existsSync(fullPath)).toBeTruthy();
        });
        it("- 파일 유무 : src/inline/all/pre_p3.html", () => {
            const fullPath = path.join(dirname, "src/inline/all/pre_p3.html");
            expect(fs.existsSync(fullPath)).toBeTruthy();
        });
        it("- 파일 유무 : src/inline/arg1_1/arg1_2/group_arg2/pre_p1_suf.html", () => {
            const fullPath = path.join(dirname, "src/inline/arg1_1/arg1_2/group_arg2/pre_p1_suf.html");
            expect(fs.existsSync(fullPath)).toBeTruthy();
        });
        it("- 파일 유무 : src/inline/arg1_1/arg1_2/group_arg2/pre_p2_suf.html", () => {
            const fullPath = path.join(dirname, "src/inline/arg1_1/arg1_2/group_arg2/pre_p2_suf.html");
            expect(fs.existsSync(fullPath)).toBeTruthy();
        });
        it("- 파일 유무 : src/inline/new_p1.html", () => {
            const fullPath = path.join(dirname, "src/inline/new_p1.html");
            expect(fs.existsSync(fullPath)).toBeTruthy();
        });
        it("- 파일 유무 : src/inline2/def1/group_def2/p1.html", () => {
            const fullPath = path.join(dirname, "src/inline2/def1/group_def2/p1.html");
            expect(fs.existsSync(fullPath)).toBeTruthy();
        });
        it("- 파일 유무 : src/inline2/def1/group_def2/p2.html", () => {
            const fullPath = path.join(dirname, "src/inline2/def1/group_def2/p2.html");
            expect(fs.existsSync(fullPath)).toBeTruthy();
        });
    });
    
    describe("< ready() 생성파일 검사 >", ()=>{
        it("- 파일 유무 : src/ready/pre_p1_suf.html", () => {
            const fullPath = path.join(dirname, "src/ready/pre_p1_suf.html");
            expect(fs.existsSync(fullPath)).toBeTruthy();
        });
        it("- 파일 유무 : src/ready/pre_p2_suf.html", () => {
            const fullPath = path.join(dirname, "src/ready/pre_p2_suf.html");
            expect(fs.existsSync(fullPath)).toBeTruthy();
        });
        it("- 파일 유무 : src/ready/pre_p3_suf.html", () => {
            const fullPath = path.join(dirname, "src/ready/pre_p3_suf.html");
            expect(fs.existsSync(fullPath)).toBeTruthy();
        });
        it("- 파일 유무 : src/ready/arg1/group_arg2/pre_p1_suf.html", () => {
            const fullPath = path.join(dirname, "src/ready/arg1/group_arg2/pre_p1_suf.html");
            expect(fs.existsSync(fullPath)).toBeTruthy();
        });
        it("- 파일 유무 : src/ready/arg1/group_arg2/pre_p2_suf.html", () => {
            const fullPath = path.join(dirname, "src/ready/arg1/group_arg2/pre_p2_suf.html");
            expect(fs.existsSync(fullPath)).toBeTruthy();
        });
        it("- 파일 유무 : src/p3.html", () => {
            const fullPath = path.join(dirname, "src/p3.html");
            expect(fs.existsSync(fullPath)).toBeTruthy();
        });
        it("- 파일 유무 : src/ready/new_p3.html", () => {
            const fullPath = path.join(dirname, "src/ready/new_p3.html");
            expect(fs.existsSync(fullPath)).toBeTruthy();
        });
    });

    describe("< 파일 비교 >", () => {
        it("- page : p1.html", () => {
            const fullPath1 = path.join(dirname, "src/ready/pre_p1_suf.html");
            const fullPath2 = path.join(dirname, "src/ready/arg1/group_arg2/pre_p1_suf.html");
            const fullPath3 = path.join(dirname, "src/inline/all/pre_p1.html");
            const fullPath4 = path.join(dirname, "src/inline/arg1_1/arg1_2/group_arg2/pre_p1_suf.html");
            const fullPath5 = path.join(dirname, "src/inline/new_p1.html");
            const data1 = fs.readFileSync(fullPath1, "utf-8");
            const data2 = fs.readFileSync(fullPath2, "utf-8");
            const data3 = fs.readFileSync(fullPath3, "utf-8");
            const data4 = fs.readFileSync(fullPath4, "utf-8");
            const data5 = fs.readFileSync(fullPath5, "utf-8");
            expect(data1 === data2 && data3 === data4 && data1 === data5).toBeTruthy();
            expect(data1).toMatchSnapshot();
        });
        it("- page : p2.html", () => {
            const fullPath1 = path.join(dirname, "src/ready/pre_p2_suf.html");
            const fullPath2 = path.join(dirname, "src/ready/arg1/group_arg2/pre_p2_suf.html");
            const fullPath3 = path.join(dirname, "src/inline/all/pre_p2.html");
            const fullPath4 = path.join(dirname, "src/inline/arg1_1/arg1_2/group_arg2/pre_p2_suf.html");
            const data1 = fs.readFileSync(fullPath1, "utf-8");
            const data2 = fs.readFileSync(fullPath2, "utf-8");
            const data3 = fs.readFileSync(fullPath3, "utf-8");
            const data4 = fs.readFileSync(fullPath4, "utf-8");
            expect(data1 === data2 && data3 === data4 && data1 === data3).toBeTruthy();
            expect(data1).toMatchSnapshot();
        });
        it("- page : p3.html", () => {
            const fullPath1 = path.join(dirname, "src/inline/all/pre_p3.html");
            const fullPath2 = path.join(dirname, "src/ready/pre_p3_suf.html");
            const fullPath3 = path.join(dirname, "src/p3.html");
            const fullPath4 = path.join(dirname, "src/ready/new_p3.html");
            const data1 = fs.readFileSync(fullPath1, "utf-8");
            const data2 = fs.readFileSync(fullPath2, "utf-8");
            const data3 = fs.readFileSync(fullPath3, "utf-8");
            const data4 = fs.readFileSync(fullPath4, "utf-8");
            expect(data1 === data2 && data3 === data4 && data1 === data3).toBeTruthy();
            expect(data1).toMatchSnapshot();
        });
    });
});
