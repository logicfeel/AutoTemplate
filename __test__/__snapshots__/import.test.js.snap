// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`task :: publish < *.hbs 검사 > - 파일 비교 : src/one.html 1`] = `
"<one><!--src/one.html-->

    # ns part 삽입
    <footer><!--part/inc/footer-->
        <p>기본 footer. </p>
        $entity.Table=1$
    </footer>
    

    # ns part 삽입
    <header><!--part/inc/header-->
      # data 삽입 
      $entity.Tables.[0]:10$
    
    
      # part 삽입
      <content><!--part/inc/content-->
          내용임
      </content></header>

    # ns helper 삽입
    $<div class="mybold">BOLD</div>$

    # ns page 삽입
    <p1><!--page/p1.html-->
        기본 페이지
    </p1>

    # ns data 조회
    $entity.Tables=10,20$

</one>

"
`;

exports[`task :: publish < 파일 비교 > - page : p1.html 1`] = `
"<p1><!--page/p1.html-->
    기본 페이지
</p1>"
`;

exports[`task :: publish < 파일 비교 > - page : p2.html 1`] = `
"<p2><!--page/p2.html-->
    # footer 조각 가져오기
    <footer><!--part/inc/footer-->
        <p>기본 footer. </p>
        $entity.Table=1$
    </footer>

    # helper 가져오기
    $<div class="mybold">BOLD</div>$
</p2>"
`;

exports[`task :: publish < 파일 비교 > - page : p3.html 1`] = `
"<p3><!--page/p3.html-->
    # page p1 가져오기
    <p1><!--page/p1.html-->
        기본 페이지
    </p1>
</p3>
"
`;
